### PR TITLE
Test authz config file with RBAC and anonymous auth

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind bug
/kind test

#### What this PR does / why we need it:

* Adds a test for authz config working with anonymous auth and default RBAC policies
* Adds ability to test authz config with local-up-cluster
* Fixes a bug inspecting --authorization-mode flag value for anonymous auth when using --authorization-config
* Fixes a bug inspecting --authorization-mode flag value for default RBAC policy when using --authorization-config

```release-note
NONE
```

/assign @enj @palnabarun @ritazh 
/milestone v1.29

### CODE REVIEW

File: test/integration/auth/authz_config_test.go [hunk 1]
```diff
@@ -63,6 +65,16 @@ authorizers:
 	)
 	t.Cleanup(server.TearDownFn)
 
+	// Make sure anonymous requests work
+	anonymousClient := clientset.NewForConfigOrDie(rest.AnonymousClientConfig(server.ClientConfig))
+	healthzResult, err := anonymousClient.DiscoveryClient.RESTClient().Get().AbsPath("/healthz").Do(context.TODO()).Raw()
+	if !bytes.Equal(healthzResult, []byte(`ok`)) {
+		t.Fatalf("expected 'ok', got %s", string(healthzResult))
```
Comment (ritazh) [inline test/integration/auth/authz_config_test.go:72]: anonymous request will hit this https://github.com/kubernetes/kubernetes/pull/121818/files#diff-15060c795e56fc5b2cc6584cac817d93dc0993f46f6b8eb3ecb57b77d6cd830aR92-R94 code path right? is there a way to test `authzmodes.ModeAlwaysAllow`

File: pkg/controlplane/apiserver/config.go [hunk 2]
```diff
@@ -169,24 +170,34 @@ func BuildGenericConfig(
 }
 
 // BuildAuthorizer constructs the authorizer. If authorization is not set in s, it returns nil, nil, nil
```
Comment (ritazh) [inline pkg/controlplane/apiserver/config.go:172]: nit: it returns nil, nil, false, nil

File: test/integration/auth/authz_config_test.go (hunk 1 above)
Comment (liggitt) [reply test/integration/auth/authz_config_test.go:72]: AlwaysAllow is tested in a bunch of places already

the integration test bit I added requires *both* anonymous auth and RBAC default policy to be working in order to work, so it exercises both fixes at once

File: pkg/controlplane/apiserver/config.go (hunk 2 above)
Comment (liggitt) [reply pkg/controlplane/apiserver/config.go]: fixed

**Comment by liggitt** [comment]
updated comment and fixed linter complaint

**Comment by liggitt** [comment]
(local-e2e was already failing, see job history)

**Pull request merged**